internal/api/api.go

```diff
@@ -36,15 +37,15 @@ type IdlerAPI interface {
 
 type idler struct {
 	openShiftClient client.OpenShiftClient
-	controller      openshift.Controller
+	controller      openshift.ControllerI
```

Comment:
ControllerI? Did you rename the Controller interface to ControllerI? I am not really happy with this one. Let's stick to Controller as interface and try finding another name for what used to be OpenShiftController. We can keep 'Controller' as interface name, since this is what's actually exposed everywhere. And the actual implementation could be "controller". It does not even have to be exported. Or controllerImpl.